client/controller/models/certs/certs_test.go

```diff
@@ -271,6 +275,15 @@ func TestCertAttach(t *testing.T) {
 	if err = Attach(&client, "test-example-com", "foo.com"); err != nil {
 		t.Fatal(err)
 	}
+
+	if err = Attach(&client, "non-existent-cert", "foo.com"); err == nil {
+		t.Fatal("An Error should have resulted from the attempt to attach a non-existent cert to a valid domain")
+	}
+
+	// TODO: enable once Attach can determine if domain is valid
```

Comment:
Not usually a fan of checking in commented-out code (test or otherwise) but wanted to throw this out there: Currently, `certs.Attach` doesn't have a way to figure out if a supplied `domain` is valid, lacking an `appID` to issue a GET request... suggestions on how it might check?